crates/rpc/rpc/src/otterscan.rs

```diff
@@ -71,11 +74,55 @@ where
     /// Handler for `getBlockTransactions`
     async fn get_block_transactions(
         &self,
-        _block_number: BlockNumberOrTag,
-        _page_number: usize,
-        _page_size: usize,
+        block_number: BlockNumberOrTag,
+        page_number: usize,
+        page_size: usize,
     ) -> RpcResult<OtsBlockTransactions> {
-        Err(internal_rpc_err("unimplemented"))
+        let block = self
+            .eth
+            .block_by_number(block_number, true)
+            .await?
+            .ok_or_else(|| internal_rpc_err("block not found"))?;
+
+        let BlockTransactions::Full(transactions) = &block.transactions else {
+            return Err(internal_rpc_err("block is not full"));
+        };
+
+        let mut transactions = transactions.clone();
+
+        // The input field returns only the 4 bytes method selector instead of the entire
+        // calldata byte blob.
+        for tx in &mut transactions {
+            tx.input = tx.input.slice(..4);
+        }
+
+        let mut receipts =
+            self.eth.block_receipts(BlockId::Number(block_number)).await?.unwrap_or(vec![]);
+
+        // For receipts
+        //  1. logs attribute returns null
+        //  2. logsBloom attribute returns null.
+        receipts.iter_mut().for_each(|receipt| {
+            receipt.logs = vec![];
+            receipt.logs_bloom = Bloom::default();
```

Comment:
hmm, I think we actually we a special ots tx receipt type here that simply excludes those fields

Reply:
I checked the reference implementation and api doc, these fields are returned by methods like `ots_searchTransactionsBefore`.  `ots_getBlockTransactions` is more like a optimized version to get a bunch of transactions and its related receipt, so it could prune some fields. Is it beneficial to define a special ots tx receipt type only used here?

Reply:
> is it beneficial to define a special ots tx receipt type only used here?

tbh I think we can just skip this step and keep the receipts included in the response